### ISSUE TITLE
Fix FP on S3757 when using Number wrapper

### ISSUE THREAD

**Issue by guillaume-dequenne-sonarsource** [SonarSource/SonarJS#2411]
S3757 should not raise on the following snippet:

```
class SomeClass {
     parse(something, index) { // eslint-disable-line no-unused-vars
        if ((typeof something === 'number' || something instanceof Number) && !isFinite(+something)) { // FP on +something
            console.log("hello");
        }
    }
}
```
Here `something` seems to be inferred of type `Number` which is considered as an `Object` and thus an issue is mistakenly reported.
